applications/client/controllers/proc_harness.py

```diff
@@ -0,0 +1,112 @@
+#########################################################################
+## Define API for interacting with procedure table
+## These methods will be used in the controller, called by the code
+## that actually performs the sync to interact with the tables
+#########################################################################
+
+from gluon import current
+
+import requests
+import json
+
+proc_table = db.procedure
+settings_table = db.client_setting
+
+def do_procedure_sync():
+    """
+    Perform sync of procedures according to the following protocol:
+        Get all timestamps from server for last stable save of all procedures associated with this device
+        Compare timestamps to timestamps retrieved at last synch and get new procedures
+        Send list of procedure IDs that are needed from server
+        Get updated data from server and save it to local database
+    """
+
+    # Get device id from settings table
+    device_id = db().select(settings_table.device_id).first().device_id
+
+    # Get authorization from server for request???
+    #   Waiting for other team to implement this method
+    #   Not sure what to do here if anything
+
+    # Request dictionary {procedure_id: last_updated_date} from server
+    server_url = myconf.get('server.host')
+    call_url = server_url + '/proc_harness/get_procedure_status(' + str(device_id) + ').json'
+    server_status = json.loads(requests.get(call_url))
+
+    # Get corresponding dictionary from client procedure table
+    client_status = get_procedure_status()
+
+    # Compare two dicts to get new procedures or procedures that have been updated
+    synch_ids = compare_dates(server_status, client_status)
+
+    # Request full procedure data for new and updated procedures from server
+    call_url = server_url + '/proc_harness/get_procedure_update(' + json.dumps(synch_ids) + '.json'
+    synch_data = json.loads(requests.get(call_url))
+
+    # Update local data
+    insert_new_procedure(synch_data)
+
+
+def get_procedure_status():
+    """
+    Returns full dictionary of format {procedure_id: last_updated_date} for all procedures on the client
+    last_updated_date comes from server so can be directly compared to dates that come from server
+
```

Comment:
We should in any case keep all dates in UTC, but I see what you mean, equal dates means same version.